app/controllers/concerns/foreman/controller/registration.rb

```diff
@@ -0,0 +1,73 @@
+module Foreman::Controller::Registration
+  extend ActiveSupport::Concern
+
+  def find_global_registration
+    template_name = Setting[:default_global_registration_item]
+    @provisioning_template = ProvisioningTemplate.unscoped.find_by(name: template_name)
+    @global_registration_vars = global_registration_vars if @provisioning_template
+  end
+
+  private
+
+  def global_registration_vars
+    permitted = Foreman::Plugin.all
+                               .map(&:allowed_registration_vars)
+                               .flatten.compact.uniq
+
+    organization = Organization.authorized(:view_organizations).find(params['organization_id']) if params['organization_id'].present?
+    location = Location.authorized(:view_locations).find(params['location_id']) if params['location_id'].present?
+    host_group = Hostgroup.authorized(:view_hostgroups).find(params['hostgroup_id']) if params["hostgroup_id"].present?
+    operatingsystem = Operatingsystem.authorized(:view_operatingsystems).find(params['operatingsystem_id']) if params["operatingsystem_id"].present?
+
+    {
+      user: User.current,
+      auth_token: User.current.jwt_token!(expiration: 4.hours.to_i),
+      organization: organization,
+      location: location,
+      hostgroup: host_group,
+      operatingsystem: operatingsystem,
+    }.merge(params.permit(permitted).to_h.symbolize_keys)
```

Comment:
can we switch the merge order so the required list always has higher priority?

e.g.
```ruby
(params.permit(permitted).to_h.symbolize_keys,merge({
￼      user: User.current,
￼      auth_token: User.current.jwt_token!(expiration: 4.hours.to_i),
￼      organization: organization,
￼      location: location,
￼      hostgroup: host_group,
￼      operatingsystem: operatingsystem,
￼    })
```

this will be safer, e.g. if plugin registers "user" accidentaly